packages/node/src/behavior/state/validation/conformance-compiler.ts

```diff
@@ -582,7 +586,7 @@ export function astToFunction(schema: ValueModel, supervisor: RootSupervisor): V
 
     function disallowEnumValue(schema: Schema): EnumMemberValidator {
         return location => {
-            throw new ConformanceError(
+            throw new EnumValueConformanceError(
```

Comment:
The use of disallow() in the memberValidator generated in the Code.Evaluate case of addEnumMemberValidation should also probably be changed to this error

That code path is activated if there is field-value-dependent conformance

Previously it did not throw as useful of an error as the static case. But now I think it's actually incorrect because it would still result in InvalidAction